### ISSUE TITLE
How to access current user ID from a Service?

### ISSUE THREAD

**Issue by gbrl** [neroniaky/angular2-token#34]
I have two models, User and Profile, and I want my Profile Service to be able to call the API using the current logged-in users' ID. I can't figure out how to do this. Any advice is appreciated. Here is my service:

    import { Injectable } from '@angular/core';
    import { Http, Headers, RequestOptions } from '@angular/http';

    @Injectable()
    export class ProfileService {

      constructor(private http: Http) {}

      find() {
        return this.http.get('/api/user/22/profile.json');
      }

    }

Basically, I need 

    'api/user/22/profile.json'

to behave like 

    'api/user/' +  user.id + '/profile.json

**Comment by neroniaky** [comment]
Hey @gbrl!
If you are accessing private resources (something only your user should have access to) you are usually not letting the front end set the user ID. Thats why devise token auth gives you the `current_user` variable which refers the currently signed in user.
So what you do is call `api/user/profile.json` with your angular service and let your backend return the right profile.

Hope this helps 😄 .

**Comment by gbrl** [comment]
Ah! That makes so much sense. I will try it out.

**Comment by MonikaMahanthappa** [comment]
@neroniaky @gbrl This question and the answer was really helpful. Thanks.  

Just to add to it Rails devise_token_auth gives `current_api_user` not the `current_user`. Minor correction. Anyway, you do make the right point.

**Comment by neroniaky** [comment]
Hey @MonikaMahanthappa, Thanks!

The current_user variable depends on the namespace you're defining the devise_token_auth model in. So in your case its the current_`api_`user because you define your user model in the `api` namespace (I guess).
